src/bin/tpch_table_generator.cpp

```diff
@@ -3,10 +3,16 @@
 #include "storage/storage_manager.hpp"
 #include "tpch/tpch_table_generator.hpp"
 
-int main() {
+#include "utils/assert.hpp"
```

Comment:
Can we generate the CSV here and compare it to that of dbgen? Not as a test, just once?

Reply:
```
#!/usr/bin/python3

import csv

for table in ['customer', 'lineitem', 'nation', 'orders', 'part', 'partsupp', 'region', 'supplier']:
    hyrise_filepath = table + '.csv'  
    dbgen_filepath = 'tpch-dbgen/' + table + '.tbl'
    with open(hyrise_filepath) as hyrise_file:
        hyrise_reader = csv.reader(hyrise_file)

        with open(dbgen_filepath) as dbgen_file:  
           dbgen_line = dbgen_file.readline().strip()
           cnt = 1
           
           for hyrise_values in hyrise_reader:
               dbgen_values = dbgen_line.split('|')
               for hyrise_value, dbgen_value in zip(hyrise_values, dbgen_values):
                   if(hyrise_value != dbgen_value and abs(float(hyrise_value) - float(dbgen_value))) > .1:
                       print("Table " + table + ", Line " + str(cnt) + ": Mismatch " + hyrise_value + " and " + dbgen_value)

               dbgen_line = dbgen_file.readline().strip()
               cnt += 1
           assert(not dbgen_line)
    print(table + " looks good")
```

SFs 0.1 and 1 are fine, running 200 over the night.

Reply:
```
customer looks good
lineitem looks good
nation looks good
orders looks good
part looks good
partsupp looks good
region looks good
supplier looks good
```